From 1c53ea3aae4f1b72420e477260551a2d9e44c086 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Tue, 17 Jul 2018 04:57:39 +0000
Subject: [PATCH 1874/2367] drm/i915/gvt: refine the initial ddb

Currently with the plane restriction patches, both SOS and UOS will not
use the cursor plane, so that we can remove the ddb allocation for the
cursor plane, and thus save some ddb for other planes.

With this patch, SOS and UOS now can support upto 4k@30hz monitors.

Signed-off-by: Min He <min.he@intel.com>
---
 drivers/gpu/drm/i915/gvt/gvt.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/gvt.c b/drivers/gpu/drm/i915/gvt/gvt.c
index 8430315..7fecfa7 100644
--- a/drivers/gpu/drm/i915/gvt/gvt.c
+++ b/drivers/gpu/drm/i915/gvt/gvt.c
@@ -210,15 +210,13 @@ static void intel_gvt_init_ddb(struct intel_gvt *gvt)
 	for_each_pipe(dev_priv, pipe) {
 		start = pipe * ddb_size / INTEL_INFO(dev_priv)->num_pipes;
 		end = start + pipe_size;
-		ddb->plane[pipe][PLANE_CURSOR].start = end - 8;
-		ddb->plane[pipe][PLANE_CURSOR].end = end;
 
 		plane_cnt = (INTEL_INFO(dev_priv)->num_sprites[pipe] + 1);
-		plane_size = (pipe_size - 8) / plane_cnt;
+		plane_size = pipe_size / plane_cnt;
 
 		for_each_universal_plane(dev_priv, pipe, plane) {
 			ddb->plane[pipe][plane].start = start +
-				(plane * (pipe_size - 8) / plane_cnt);
+				(plane * pipe_size / plane_cnt);
 			ddb->plane[pipe][plane].end =
 				ddb->plane[pipe][plane].start + plane_size;
 		}
-- 
2.7.4

